### PR TITLE
zfs: various fixes

### PR DESCRIPTION
**Pull request by 0xC0ncord**
Fixes for ZFS, including allowing Zed to use sendmail and write LED statuses to enclosure devices and for the `zpool` utility to properly create and destroy ZFS pools.

Also add file contexts for nullmailer in `/var/spool` and allow mailer domains to trigger it.

### CODE REVIEW

**Comment by 0xC0ncord** [comment]
I am not set on the type for `/var/spool/nullmailer/trigger`. `mqueue_spool_t` may be more optimal.

File: policy/modules/services/zfs.te [hunk 1]
```diff
@@ -112,9 +122,35 @@ mount_exec(zfs_t)
...
+
+	manage_dirs_pattern(zed_t, zed_mail_tmp_t, zed_mail_tmp_t)
+	manage_files_pattern(zed_t, zed_mail_tmp_t, zed_mail_tmp_t)
+	files_tmp_filetrans(zed_t, zed_mail_tmp_t, { dir file })
+
+	mta_sendmail_domtrans(zed_t, zed_mail_t)
+
+	allow zed_mail_t self:capability { dac_override dac_read_search };
+
+	storage_raw_rw_fixed_disk(zed_mail_t)
```
Comment (pebenito) [inline policy/modules/services/zfs.te:155]: This is very concerning, a networked process with raw disk access.

Comment (0xC0ncord) [reply policy/modules/services/zfs.te:155]: I'll see if there's a workaround or some other way we can avoid this.

Comment (pebenito) [reply policy/modules/services/zfs.te:155]: If not, please make a tunable. Then it will at least make the users aware of the access.

Comment (0xC0ncord) [reply policy/modules/services/zfs.te:155]: That was my next question. :)

Comment (0xC0ncord) [reply policy/modules/services/zfs.te]: I found that this access is not needed at all yet for some reason is attempted by the `mail` process that gets spawned. I ended up just `dontaudit`ing it.

Comment (pebenito) [reply policy/modules/services/zfs.te]: Perhaps a leaked fd?

**Pull request merged**